apps/projects/routes.py

```diff
@@ -176,100 +377,814 @@ def get(self, project_id):
+                    type: integer
+                    example: 5
+                  end:
+                    type: integer
+                    example: 10
+              crop:
+                type: object
+                properties:
+                  width:
+                    type: integer
+                    example: 480
+                  height:
+                    type: integer
+                    example: 360
+                  x:
+                    type: integer
+                    example: 10
+                  y:
+                    type: integer
+                    example: 10
+              rotate:
+                type: object
+                properties:
+                  degree:
+                    type: integer
+                    example: 90
+        responses:
+          200:
+            description: OK
+            schema:
+              type: object
+              properties:
+                filename:
+                  type: string
+                  example: fa5079a38e0a4197864aa2ccb07f3bea_v2.mp4
+                url:
+                  type: string
+                  example: https://example.com/url_raw/fa5079a38e0a4197864aa2ccb07f3bea
+                storage_id:
+                  type: string
+                  example: 2019/5/fa5079a38e0a4197864aa2ccb07f3bea.mp4
+                metadata:
+                  type: object
+                  example: {}
+                mime_type:
+                  type: string
+                  example: video/mp4
+                create_time:
+                  type: object
+                  properties:
+                    $date:
+                      type: integer
+                      example: 1556853105063
+                original_filename:
+                  type: string
+                  example: video.mp4
+                client_info:
+                  type: string
+                  example: Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:66.0) Gecko/20100101 Firefox/66.0
+                version:
+                  type: integer
+                  example: 2
+                parent:
+                  type: object
+                  parameters:
+                    _id:
+                      type: object
+                      parameters:
+                        $oid:
+                          type: string
+                          example: 5ccbc4104dfd9b8fa153d60e
+                processing:
+                  type: boolean
+                  example: False
+                thumbnails:
+                  type: object
+                  example: {}
+                _id:
+                  type: object
+                  properties:
+                    $oid:
+                      type: string
+                      example: 5cbd5acfe24f6045607e51aa
         """
+        # validate user-agent
+        user_agent = check_user_agent()
+        # validate request
+        check_request_schema_validity(request.get_json(), self.SCHEMA_EDIT)
```

Comment:
I'm not sure if it's good idea to call `request.get_json()` all the time when you need it, just save it to the variable and reuse.